src/model.ts

```diff
@@ -919,14 +919,18 @@ export class GitExtension implements IGitExtension {
     return await this._taskHandler.execute<Git.ILogResult>(
       'git:fetch:log',
       async () => {
-        return await requestAPI<Git.ILogResult>(
-          URLExt.join(path, 'log'),
-          'POST',
-          {
-            history_count: count,
-            follow_path: this.selectedHistoryFile?.to
-          }
-        );
+        try {
+          return await requestAPI<Git.ILogResult>(
+            URLExt.join(path, 'log'),
+            'POST',
+            {
+              history_count: count,
+              follow_path: this.selectedHistoryFile?.to
+            }
+          );
+        } catch (error) {
+          return { code: 0, commits: [] };
```

Comment:
As the code in the panel is:

https://github.com/jupyterlab/jupyterlab-git/blob/c7e7363a5e749ea4590984f3a0303694f25212aa/src/components/GitPanel.tsx#L264-L270

The best is to return a non-null code (`1` e.g) and the state will be an empty array as wanted.

Reply:
Pushed this change! Thanks for the guidance @fcollonval!